### PR TITLE
Creating virtual environment for each test

### PR DESCRIPTION
**Pull request by sungeunbae**
Creates virtual environment at /tmp/qcore/#hash#, installs requirements.
Runs test, then removes after.

Docker images all removed from the server to release space.
Binaries are kept under /nesi/project/nesi00213/tools 

Note: For virtual environments, pytest and pytest-black should be included in requirements.txt for the auto-tests to run

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,30 +1,47 @@
...
-		docker images sungeunbae/qcore-ubuntu-minimal -q
-		"""
-	    }
-	}
+        stage('Settin up env') {
+            steps {
+                echo "Start virtual environment"   
+                sh """ 
+                    pwd
+                    env
```
Comment (jasonmotha) [inline Jenkinsfile:9]: If we're using these as debug output for the logs, it would be nice to wrap it with "Current working directory:" etc.

**Review by jasonmotha — CHANGES_REQUESTED** [review]
@jpa198 Thoughts about compiling emod3d-tools as part of environment setup. Just because the binaries may change and hence break the tests?

**Comment by jpa198** [comment]
> @jpa198 Thoughts about compiling emod3d-tools as part of environment setup. Just because the binaries may change and hence break the tests?

I think pulling EMOD3D and compiling every time we do a test is too much. I think the responsibility for updating the binaries should be on the EMOD3D repo. 
So that probably needs a jenkins file to build the tools on PR creation or merge.
EMOD3D is normally low traffic, so it shouldn't be an issue too often.

**Comment by jasonmotha** [comment]
> > @jpa198 Thoughts about compiling emod3d-tools as part of environment setup. Just because the binaries may change and hence break the tests?
> 
> I think pulling EMOD3D and compiling every time we do a test is too much. I think the responsibility for updating the binaries should be on the EMOD3D repo.
> So that probably needs a jenkins file to build the tools on PR creation or merge.
> EMOD3D is normally low traffic, so it shouldn't be an issue too often.

Sweet. Agreed with all your thoughts.

File: Jenkinsfile (hunk 1 above)
Comment (sungeunbae) [reply Jenkinsfile:9]: You mean as a separate "stage"? Sure - can do.

**Comment by sungeunbae** [comment]
> > > @jpa198 Thoughts about compiling emod3d-tools as part of environment setup. Just because the binaries may change and hence break the tests?
> > 
> > 
> > I think pulling EMOD3D and compiling every time we do a test is too much. I think the responsibility for updating the binaries should be on the EMOD3D repo.
> > So that probably needs a jenkins file to build the tools on PR creation or merge.
> > EMOD3D is normally low traffic, so it shouldn't be an issue too often.
> 
> Sweet. Agreed with all your thoughts.

I only managed to build a couple of tools on ec2 - had to temporarily install openmpi and FFTW3, and it crashed at Fractal3D. At the point, I had all the binaries needed for the test, so I didn't go further to fix the build issues.

File: Jenkinsfile (hunk 1 above)
Comment (sungeunbae) [reply Jenkinsfile:9]: Done in the latest patch

File: Jenkinsfile [hunk 2]
```diff
@@ -1,19 +1,26 @@
...
                     pwd
                     env
+                """
+            }
+        }
+        stage('Settin up env') {
+            steps {
+                echo "Start virtual environment"   
+                sh """
+# Each stage needs custom setting done again. By default /biin/python is used.
```
Comment (jasonmotha) [inline Jenkinsfile:17]: /bin/python*

File: Jenkinsfile [hunk 3]
```diff
@@ -1,19 +1,26 @@
 pipeline {
     agent any 
     stages {
-        stage('Settin up env') {
+        stage('Checking env') {
             steps {
-                echo "Start virtual environment"   
-                sh """ 
+                echo "Check the environment"
```
Comment (jasonmotha) [inline Jenkinsfile:6]: I'd still like some labelling on the lines below. Just so as a user you can tell what each path is from the logfile.

Comment (sungeunbae) [reply Jenkinsfile]: I misunderstood what you meant. Reverted the previous change (separate check stage) and added more comments

**Review by jasonmotha — APPROVED** [review]
Cool. Looks good.

**Review by jpa198 — APPROVED** [review]

**Pull request merged**